src/components/NoteDetails/NoteDetailsReducer.js

```diff
@@ -1,22 +1,39 @@
-import { EDITOR_NOTE_NAME_CHANGED_ACTION } from 'components/NoteDetails/NoteDetailsActions';
-import { CHANGE_SELECTED_NOTE_ACTION } from 'components/NotesMindMap/NotesMindMapActions';
+import {
+  EDITOR_NOTE_NAME_CHANGED_ACTION,
+  EDITOR_NOTE_CONTENT_CHANGED_ACTION,
+} from 'components/NoteDetails/NoteDetailsActions';
+import {
+  CHANGE_SELECTED_NOTE_ACTION,
+  CHANGE_NOTE_TEXT_ACTION,
+} from 'components/NotesMindMap/NotesMindMapActions';
 
 const defaultState = {
   noteName: '',
+  noteText: '',
 };
 
 export const noteDetailsReducer = (state = defaultState, { type, data }) => {
   switch (type) {
     case EDITOR_NOTE_NAME_CHANGED_ACTION:
       return {
         ...state,
-        noteName: data.value,
+        noteName: data.newNoteName,
       };
     case CHANGE_SELECTED_NOTE_ACTION:
       return {
         ...state,
         noteName: data.note.name,
       };
+    case EDITOR_NOTE_CONTENT_CHANGED_ACTION:
+      return {
+        ...state,
+        noteText: data.noteText,
+      };
+    case CHANGE_NOTE_TEXT_ACTION:
+      return {
+        ...state,
+        noteText: data,
+      };
```

Comment:
Looks like `EDITOR_NOTE_CONTENT_CHANGED_ACTION` and `CHANGE_NOTE_TEXT_ACTION` do the same thing: they both return `{...state, noteText}`.

Why do we need both events? What is the difference between them?

Please use one name everywhere: `note_content` or `note_text`, in order to avoid confusion.

Reply:
I need to think, can't answer right now.

Reply:
> Looks like `EDITOR_NOTE_CONTENT_CHANGED_ACTION` and `CHANGE_NOTE_TEXT_ACTION` do the same thing: they both return `{...state, noteText}`.
> 
> Why do we need both events? What is the difference between them?

I don't know if it important, but `EDITOR_NOTE_CONTENT_CHANGED_ACTION` returns `{...state, noteText: data.noteText}` when  `CHANGE_NOTE_TEXT_ACTION` returns `{...state, noteText: data}`

Reply:
Okay I took a closer look :)

`CHANGE_NOTE_TEXT_ACTION` is coming from `NoteMindMap`. It is a badly named action.
When selected note changes, note text is requested in `NotesMindMapSagas.js`. When the note text is fetched from Google Drive, `CHANGE_NOTE_TEXT_ACTION` is dispatched.
The action name should be `note text fetched`.

So we need to keep this event, otherwise when the selected note changes, our editor won't update the text.

Reply:
So, do you want me to change name of this action to NOTE_TEXTFETCHED?

Reply:
No :) We can't do it in this PR. We should rename it later in another PR, to `NOTE_CONTENT_FETCHED`.